scripts/tank_cmd.py

```diff
@@ -1096,6 +1099,14 @@ def run_engine_cmd(log, pipeline_config_root, context_items, command, using_cwd,
             exit_code = show_help(logger)
             sys.exit(exit_code)
 
+    # If we are not authenticated.
+    if not shotgun.is_script_user_authenticated():
```

Comment:
I see where you are coming from, but at the same time I am not comfortable hiding script user functionality in the session management code.

However, it might be cleaner if I move all authentication related checks (is script user authenticated bit) and code into session.py and renamed it authenticate.py. console_login would become console_authenticate, which is an umbrela term for script user login, user/pass login or render-farm-like login.

Then console_autheticate could deal with the many ways we authenticate in Toolkit.

If you are fine with this proposal, I implement it in a separate commit at the very end as to not muddy the diffs too much.